### PR TITLE
preferences

### PR DESCRIPTION
**Pull request by seanpearsonuk**
Add preferences to PyFluent

### CODE REVIEW

File: codegen/datamodelgen.py [hunk 1]
```diff
@@ -336,7 +348,7 @@ def write_static_info(self) -> None:
                 self._write_static_info("Root", info.static_info, f)
                 doc_dir = Path(
                     _MESHING_DM_DOC_DIR
-                    if info.mode == "meshing"
+                    if "meshing" in info.modes
```
Comment (seanpearsonuk) [inline codegen/datamodelgen.py:351]: what to do?

File: codegen/datamodelgen.py [hunk 2]
```diff
@@ -345,7 +357,7 @@ def write_static_info(self) -> None:
                 self._write_doc_for_model_object(
                     info.static_info,
                     doc_dir / name,
-                    f"{info.mode}.datamodel.{name}",
+                    f"{info.modes[0]}.datamodel.{name}",
```
Comment (seanpearsonuk) [inline codegen/datamodelgen.py:360]: again, what?

File: src/ansys/fluent/core/session.py [hunk 3]
```diff
@@ -180,6 +182,26 @@ def _download(self, file_name: str, local_file_path: str = None):
             self._uploader = _Uploader(self.fluent_connection._remote_instance)
         return self._uploader.download(file_name, local_file_path)
 
+    '''
+    @property
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:186]: should at least factor out the code

File: codegen/datamodelgen.py [hunk 4]
```diff
@@ -82,9 +82,11 @@ def _build_command_docstring(name: str, info: Any, indent: str):
 
 
 class DataModelStaticInfo:
-    def __init__(self, rules: str, mode: str, version: str, rules_save_name: str = ""):
+    def __init__(
+        self, rules: str, modes: tuple, version: str, rules_save_name: str = ""
+    ):
         self.rules = rules
-        self.mode = mode
+        self.modes = modes
```
Comment (seanpearsonuk) [inline codegen/datamodelgen.py:89]: using a tuple now because preferences is not specific to a single mode

File: codegen/datamodelgen.py [hunk 5]
```diff
@@ -334,21 +351,28 @@ def write_static_info(self) -> None:
...
+                            doc_dir / name,
+                            f"{mode}.datamodel.{name}",
+                            f"ansys.fluent.core.datamodel_{self.version}.{name}",
+                            "Root",
+                        )
+                        """
+                        other instance in api/core/meshing/datamodel/preferences/index,
+                        use :noindex: for one of them
+                        """
+                        break
```
Comment (seanpearsonuk) [inline codegen/datamodelgen.py:375]: Need to address this with :noindex:

**Review by hpohekar — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

File: codegen/datamodelgen.py [hunk 6]
```diff
@@ -105,16 +107,19 @@ class DataModelGenerator:
...
-                "PMFileManagement", "meshing", self.version
+                "PMFileManagement", ("meshing",), self.version
             ),
             "icing": DataModelStaticInfo(
-                "flserver", "flicing", self.version, "flicing"
+                "flserver", ("flicing",), self.version, "flicing"
+            ),
+            "preferences": DataModelStaticInfo(
+                "preferences", ("meshing", "solver", "flicing,"), self.version
             ),
```
Comment (seanpearsonuk) [inline codegen/datamodelgen.py:123]: @hpohekar here we will add an entry for solver_workflow with modes = ("solver",)

File: src/ansys/fluent/core/session_solver.py [hunk 7]
```diff
@@ -99,3 +100,10 @@ def parallel(self):
     def report(self):
         """report settings."""
         return self._root.report
+
+    @property
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session_solver.py:104]: @hpohekar we need to add similar for solver_workflow

File: src/ansys/fluent/core/session.py [hunk 8]
```diff
@@ -32,6 +32,16 @@ def parse_server_info_file(filename: str):
     return ip, port, password
 
 
+def _get_preferences(session):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:35]: @hpohekar it's possible that this is part of a repeating pattern that could be factored out

**Pull request merged**